sw/simulator/nps/nps_autopilot_rover.c

```diff
@@ -210,3 +228,43 @@ void sim_overwrite_ins(void)
   // Here we make sure that ENU states are recalculated
   stateCalcPositionEnu_i();
 }
+
+
+#ifdef NPS_BYPASS_LIDAR
+// If you want to use the lidar simulation, you need to use the lidar correction module
+void sim_overwrite_lidar(void)
```

Comment:
I don't think NPS is the right place for this, especially this file.
It seems to me that what you do is either use the real lidar or make a simulation. So in the end, what you need is a simulation mode directly from the lidar driver (or your obstacle module, I'm not sure).
You have to options for that:
- make a dedicated simulation module that you load only in the `nps` target (and the tfmini in the `ap` section)
- use the `target` attribute of the existing driver to compile different files for `ap` and `nps` targets.